src/main/scala/akka/persistence/jdbc/query/JournalSequenceActor.scala

```diff
@@ -107,6 +94,69 @@ class JournalSequenceActor(readJournalDao: ReadJournalDao, config: JournalSequen
       context.become(receive(currentMaxOrdering, missingByCounter, moduloCounter, newDelay))
   }
 
+  /**
+   * This method that implements the "find gaps" algo. It's the meat and main purpose of this actor.
+   */
+  def findGaps(elements: Seq[OrderingId], currentMaxOrdering: OrderingId, missingByCounter: Map[Int, Set[OrderingId]], moduloCounter: Int) = {
+
+    // list of elements that will be considered as genuine gaps.
+    // `givenUp` is whether empty or is was filled on a previous iteration
+    val givenUp = missingByCounter.getOrElse(moduloCounter, Set.empty)
+
+    val (nextMax, _, missingElems) =
+      // using the ordering elements that were fetched, we verify if there are any gaps
+      elements.foldLeft[(OrderingId, OrderingId, Set[OrderingId])](currentMaxOrdering, currentMaxOrdering, Set.empty) {
+
+        case ((currentMax, previousElement, missing), currentElement) =>
+
+          // we accumulate in newMissing the gaps we detect on each iteration
+          val newMissing = currentElement match {
+
+            // if current element is contiguous to previous, there is no gap
+            case e if e == previousElement + 1 => missing
+
+            // if it's a gap and has been detected before on a previous iteration we give up
+            // that means that we consider it a genuine gap that will never be filled
+            case e if givenUp(e) => missing
+
+            // any other case is a gap that we expect to be filled soon
+            case _ =>
+              val currentlyMissing = previousElement + 1 until currentElement
+              // we don't want to declare it as missing if it has been already declared on a previous iterations
+              def alreadyMissing(e: Long) = missingByCounter.values.exists(_.contains(e))
+              missing ++ currentlyMissing.filterNot(alreadyMissing)
+          }
+
+          // we must decide if the if we move the cursor forward
+          val newMax =
+            if ((currentMax + 1).until(currentElement).forall(givenUp.contains)) currentElement
+            else currentMax
```

Comment:
Previous check was:
```scala
if (currentMax + 1 == elem) elem
else if ((currentMax + 1).until(elem).forall(givenUp.contains)) elem		
else currentMax
```

The first branch is unnecessary because when `currentMax + 1 == elem`, the range will be empty and the second branch will always return true.

```scala
val a = 1
val b = 2
(a +1).until(b).forall(_ => false)
res2: Boolean = true
```
`forall` on an empty range always return true.

The only drawback of removing the first if is that it becomes less clear. Maybe we should put it back for the sake of clarity. Any thoughts?

Reply:
Looks good to me, you can add a comment for clarity if you want